### DISCUSSION TITLE
Features request

### DISCUSSION THREAD

**Discussion by psydvl** [lapce/lapce#1076] [Ideas]
1. In file/folder menu from right click
	a. Duplicate, Move (mostly same as rename, but shows absolute path)
	b. Open with default application (e.g. xdg-open on linux)
	c. Open with possible application (e.g. https://github.com/timgott/gtk-open-with on linux)
	d. Blame file
	e. Find in folder
2. Indent control: space or tabs, its size, option to convert space to tabs and vice versa
3. In current file selection menu:
	a. show diff hunk,
	b. file/lines history, blame file (maybe opens in different VCS appliaction like Sublime Merge)
4. Find and replace dialog; in file, folder, project
5. Open multiple folders in one window
6. Project and workspace: first to handle folders, specific plugins options, env variable, second - opened folder (tree status), opened files, search history, 
7. Separate tab to search (like one in Sublime Text)
8. Separate plugins and themes
9. Something like https://packagecontrol.io/ for plugins
10. Linters support like LSP (e.g. sublimelinter.com/)
11. File bookmarks (toggle via Ctrl-F2, next via F2)
12. Multiple keyboard shortcuts for option (like Alt-F4 and Ctrl-Q to close window)

PS: I'm looking for replace to Sublime Text one day, so moslty those features inspired by it

**Comment by panekj** [comment]
> 1. In file/folder menu from right click
>    a. Duplicate, Move (mostly same as rename, but shows absolute path)

Can be satisfied with `Copy`/`Cut`/`Paste`

>    b. Open with default application (e.g. xdg-open on linux)
>    c. Open with possible application (e.g. [timgott/gtk-open-with](https://github.com/timgott/gtk-open-with) on linux)

This will be possible after adding command to open URI

> 2. Indent control: space or tabs, its size, option to convert space to tabs and vice versa

Needs rework of how indentation is handled

> 3. In current file selection menu:
>    a. show diff hunk,

Can be opened via Git tab

> 5. Open multiple folders in one window

You can already do that, just create new tab

> 7. Separate tab to search (like one in Sublime Text)
> 8. Separate plugins and themes

That's already how it works

> 9. Something like [packagecontrol.io](https://packagecontrol.io/) for plugins

There is no extensions store but we already have plugins management inside editor

> 10. Linters support like LSP (e.g. sublimelinter.com/)

If they support LSP, it's already implemented

> 12. Multiple keyboard shortcuts for option (like Alt-F4 and Ctrl-Q to close window)

Already implemented

**Reply by psydvl** [reply]
> > 1. In file/folder menu from right click
> >    a. Duplicate, Move (mostly same as rename, but shows absolute path)
> 
> Can be satisfied with `Copy`/`Cut`/`Paste`

I think separate command for Move will be more useful than Copy/Paste

> > b. Open with default application (e.g. xdg-open on linux)
> > c. Open with possible application (e.g. [timgott/gtk-open-with](https://github.com/timgott/gtk-open-with) on linux)
> 
> This will be possible after adding command to open URI

Can you explain, what you mean, please?
Adding in source? Adding in new plugin?

> > 2. Indent control: space or tabs, its size, option to convert space to tabs and vice versa
> 
> Needs rework of how indentation is handled

So, you don't want this, or it's just a note?

> > 3. In current file selection menu:
> >    a. show diff hunk,
> 
> Can be opened via Git tab

Git tab shows all changes in file, while sometimes you need only few lines in big file
I expect next behavior: you're selecting text and all lines with select are showing diff

> > 5. Open multiple folders in one window
> 
> You can already do that, just create new tab

Oh, yes, just found it, but this is very hidden now: only, in command palette
And sometimes it's better to see all open files in both folders
Is it possible to change behavior and show them in the same tab?
(Look on next screenshot, there are `tdesktop` folder and `Telegram` next)

> > 7. Separate tab to search (like one in Sublime Text)
> > 8. Separate plugins and themes
> 
> That's already how it works

7. By tab, I mean exactly tab with search history:
<details>
<summary>Screenshot</summary>

![Screenshot from 2022-09-09 05-48-56](https://user-images.githubusercontent.com/43755002/189265410-bbb3b88f-e78f-4f74-9c1d-362290532fde.png)

</details>

9. No
<details>
<summary>Screenshot</summary>

![Screenshot from 2022-09-09 05-38-49](https://user-images.githubusercontent.com/43755002/189265400-b207af5b-9f7f-4d02-95e5-acedf8ae8f8c.png)

</details>

> > 9. Something like [packagecontrol.io](https://packagecontrol.io/) for plugins
> 
> There is no extensions store but we already have plugins management inside editor

Am I wrong or now listed extension can be changed only with new `Lapce` version?

> > 10. Linters support like LSP (e.g. sublimelinter.com/)
> 
> If they support LSP, it's already implemented

By linter, I mean linter, mostly they don't have LSP support
But they provide warning messages that can be shown in "Warnings"
